staff/gianluca-bragaglia/pre-course/bingo-pro-html/bingo-pro-HTML.js

```diff
@@ -0,0 +1,329 @@
+    var btnNewGame = document.getElementById("btnNewGame");
+    btnNewGame.style.display = "none";
+    var btnEndGame = document.getElementById("btnEndGame");
+    var btnStart = document.getElementById("btnStart");
+
+
+
+    btnStart.onclick = function() {
+
+        playerName = document.getElementById('playerName').value;
+        startBlock.style.display = "none";
+        card15N.innerHTML = card15.join("&nbsp;&nbsp;&nbsp;");
+        card510N.innerHTML = card510.join("&nbsp;&nbsp;&nbsp;");
+        card1015N.innerHTML = card1015.join("&nbsp;&nbsp;&nbsp;");
+        pointsN.style.display = "inline-block";
+        buttonsN.style.display = "inline-block";
+        card15N.style.display = "inline-block";
+        card1015N.style.display = "inline-block";
+        card1015N.style.display = "inline-block";
+        randDiv.style.display = "inline-block";
+        newTurn();
+
+    }
+
+    btnNewN.onclick = function() {
+        newTurn();
+    }
+
+    btnNewGame.onclick = function() {
+        document.getElementById('ranking').innerHTML = '';
+        document.getElementById('playerName').value = ' ';
+        bingo();
+    }
+
+    btnEndGame.onclick = function() {
+        buttonsN.style.display = "none";
+        randDiv.style.display = "none";
+        cardCont.style.display = "none";
+        pointsN.style.display = "none";
+        matchN.style.display = "none";
+        status = 'loose';
+        playerPoints = 0;
+        arrPlayers.push({ name: playerName, puntos: playerPoints, status: status });
+        results();
+    }
+
+
+    function addNumberCard() {
+
+        var trackingArr = [];
+        var targetCount = 15;
+        var currentCount = 0;
+        var min = 1;
+        var max = 90;
+        var rnd;
+
+        while (currentCount < targetCount) {
+            rnd = Math.floor(Math.random() * (max - min + 1)) + min;
+            if (!trackingArr[rnd]) {
+                trackingArr[rnd] = rnd;
+                card[currentCount] = rnd;
+                currentCount += 1;
+            }
+        }
+
+        return card; // Will contain 15 unique, random numbers between 1 and 90.
+
+    }
+
+
+
+    function generateArrRandom() {
+
+        var trackingArr = [];
+        var targetCount = 90;
+        var currentCount = 0;
+        var min = 1;
+        var max = 90;
+        var rnd;
+
+        while (currentCount < targetCount) {
+            rnd = Math.floor(Math.random() * (max - min + 1)) + min;
+            if (!trackingArr[rnd]) {
+                trackingArr[rnd] = rnd;
+                arrRandomNum[currentCount] = rnd;
+                currentCount += 1;
+            }
+        }
+
+        return arrRandomNum // Will contain 90 unique, random numbers between 1 and 90.
+
+
+    }
+    generateArrRandom();
+
+
+
+    function generateRandomNum() {
+
+        var numRandom = arrRandomNum[arrRandomNum.length - 1];
+        var r = arrRandomNum.length;
+        return numRandom;
```

Comment:
there is a function that you can use for this: ```Math.rand()```

Reply:
I think I can't use Math.rand() in this case, because I want the last number that are in the array arrRandomNum, not a simple random number.